### PR TITLE
added api for executing project files/folders

### PR DESCRIPTION
**Pull request by olensmar**

### CODE REVIEW

File: src/main/java/com/smartbear/readyapi/client/execution/ApiClientWrapper.java [hunk 1]
```diff
@@ -168,18 +168,20 @@ private Client getClient() {
...
-            ObjectMapper mapper = getObjectMapper();
-            mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
-            mapper.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
             if (contentType.startsWith("application/json")) {
+                ObjectMapper mapper = getObjectMapper();
+                mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+                mapper.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
                 return mapper.writeValueAsString(obj);
             }
+            else {
```
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/ApiClientWrapper.java]: formatting.

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 2]
```diff
@@ -282,4 +290,81 @@ private GenericType getReturnTypeHarLogRoot() {
...
+            } else {
+                path += "/xml";
+            }
+
+            byte[] data = Files.readAllBytes(file.toPath());
+
+            return invokeAPI(path, TestSteps.HttpMethod.POST.name(), data,
+                type, queryParams, null);
+        }
+        catch( IOException e ){
```
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: formatting

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 3]
```diff
@@ -282,4 +290,79 @@ private GenericType getReturnTypeHarLogRoot() {
...
+
+        List<Pair> queryParams = new ArrayList<>();
+        queryParams.add(new Pair("async", String.valueOf(async)));
+
+        String path = ServerDefaults.SERVICE_BASE_PATH + "/executions";
+        String type = "application/xml";
+
+        try {
+            // composite project?
+            if (file.isDirectory()) {
```
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: Should we throw an exception with user friendly message if file doesn't exist ?

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 4]
```diff
@@ -282,4 +290,79 @@ private GenericType getReturnTypeHarLogRoot() {
...
+            throw new ApiException(500, "Failed to read project; " + e.toString());
+        }
+    }
+
+    private File zipCompositeProject(File dir) throws IOException {
+        File zipFile = File.createTempFile("soapui-project", ".zip");
+
+        byte[] buffer = new byte[1024];
+
+        FileOutputStream fout = new FileOutputStream(zipFile);
```
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: you can try-with-resources here.

File: src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java [hunk 5]
```diff
@@ -282,4 +290,82 @@ private GenericType getReturnTypeHarLogRoot() {
     public void setBasePath(String basePath) {
         apiClient.setBasePath(basePath);
     }
+
+    @Override
+    public ProjectResultReport postProject(File file, boolean async, HttpBasicAuth auth) throws ApiException {
```
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: We should add parameters to specify the test suite and test case names. We support that in the server API.

Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: agreed - and perhaps also environment

Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java]: yes, that too :)

**Pull request merged**